module/VuFindSearch/src/VuFindSearch/Backend/Primo/RestConnector.php

```diff
@@ -0,0 +1,705 @@
+<?php
+
+/**
+ * Primo Central connector (REST API).
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2010.
+ * Copyright (C) The National Library of Finland 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Spencer Lamm <slamm1@swarthmore.edu>
+ * @author   Anna Headley <aheadle1@swarthmore.edu>
+ * @author   Chelsea Lobdell <clobdel1@swarthmore.edu>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Oliver Goldschmidt <o.goldschmidt@tuhh.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFindSearch\Backend\Primo;
+
+use Laminas\Session\Container as SessionContainer;
+
+use function array_key_exists;
+use function in_array;
+use function is_array;
+use function strlen;
+
+/**
+ * Primo Central connector (REST API).
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Spencer Lamm <slamm1@swarthmore.edu>
+ * @author   Anna Headley <aheadle1@swarthmore.edu>
+ * @author   Chelsea Lobdell <clobdel1@swarthmore.edu>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Oliver Goldschmidt <o.goldschmidt@tuhh.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class RestConnector implements ConnectorInterface, \Laminas\Log\LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindSearch\Backend\Feature\ConnectorCacheTrait;
+
+    /**
+     * HTTP client factory
+     *
+     * @var callable
+     */
+    protected $clientFactory;
+
+    /**
+     * Primo JWT API URL
+     *
+     * @var string
+     */
+    protected $jwtUrl;
+
+    /**
+     * Primo REST API search URL
+     *
+     * @var string
+     */
+    protected $searchUrl;
+
+    /**
+     * Institution code
+     *
+     * @var string
+     */
+    protected $inst;
+
+    /**
+     * Session container
+     *
+     * @var SessionContainer
+     */
+    protected $session;
+
+    /**
+     * Response for an empty search
+     *
+     * @var array
+     */
+    protected static $emptyQueryResponse = [
```

Comment:
There are some things like this and `$inst` that might make sense to refactor to an abstract base class... but if you're expecting that the legacy Connector class will be removed quickly, then I understand if you prefer to duplicate rather than refactor.

Reply:
That's intentional. I believe the old connector needs to be retired fairly soon after the new one goes live since it doesn't work well anymore.